src/include/sof/audio/component.h

```diff
@@ -271,6 +271,36 @@ enum comp_copy_type {
 struct comp_driver;
 struct comp_ipc_config;
 union ipc_config_specific;
+struct ipc4_module_bind_unbind;
+
+#define COMP_BIND_TYPE_SOURCE 0
+#define COMP_BIND_TYPE_SINK 1
+
+struct bind_info {
+	/* pointer to IPC4 bind data */
+	struct ipc4_module_bind_unbind *ipc4_data;
+
+	/* type of binding
+	 * bind call will be called twice for every component, first when binding a data source,
+	 * than when binding data sink.
+	 * bind_type is indicating type of binding, getting  one of values COMP_BIND_TYPE_*
+	 */
+	int bind_type;
```

Comment:
`enum`? But can it ever be anything other than source / sink? If not, maybe the original `bool link_is_source` or similar, was actually clearer. But if you prefer an `enum` no problem with that either